lib/fedora_migrate/content_mover.rb

```diff
@@ -1,24 +1,44 @@
 module FedoraMigrate
   class ContentMover < Mover
 
+    include DatastreamVerification
+
+    class Report
+      attr_accessor :name, :mime_type, :original_date, :error
+      def success?
+        error.nil?
+      end
+    end
+
     def migrate
-      return nil_content_message if source.content.nil?
+      return report if nil_source
       move_content
+      report_results
       insert_date_created_by_application
+      super
+    end
+
+    def results_report
+      Report.new
```

Comment:
When ContentMover gets initialized, `@report` is set to `results_report` which here is a Report class. When `migrate` is run, `@report` is returned. This happens in the Mover superclass.